pyplanet/apps/contrib/live_rankings/__init__.py

```diff
@@ -109,27 +109,57 @@ async def handle_scores(self, players):
+		elif 'rounds' in current_script or 'team' in current_script or \
+			'trackmania/tm_rounds_online' in current_script or 'trackmania/tm_teams_online' in current_script:
 			for player in players:
 				if 'map_points' in player:
 					if player['map_points'] != -1:
-						new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], points_added=0)
+						new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints=player['match_points'], points_added=0)
 						self.current_rankings.append(new_ranking)
 				elif 'mappoints' in player:
 					if player['mappoints'] != -1:
-						new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], points_added=0)
+						new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints=player['matchpoints'], points_added=0)
 						self.current_rankings.append(new_ranking)
-
+		elif 'cup' in current_script or 'trackmania/tm_cup_online' in current_script:
+			if section == 'EndRound':
+				mode_settings = await self.instance.mode_manager.get_settings()	
+				pointlimit = mode_settings['S_PointsLimit']
+				for player in players:
+					if 'map_points' in player:
+						if player['map_points'] != -1 and player['match_points'] == pointlimit:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints='Finalist', points_added=0)
+							self.current_rankings.append(new_ranking)
+						elif player['map_points'] != -1 and player['match_points'] > pointlimit:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints='Winner', points_added=0)
+							self.current_rankings.append(new_ranking)
+						else:
+							new_ranking = dict(login=player['player'].login, nickname=player['player'].nickname, score=player['map_points'], score_matchpoints=player['match_points'], points_added=0)
+							self.current_rankings.append(new_ranking)
+					elif 'mappoints' in player:
+						if player['mappoints'] != -1 and player['matchpoints'] == pointlimit:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints='$wFinalist', points_added=0)
+							#self.current_rankings.append(new_ranking)
+						elif player['mappoints'] != -1 and player['matchpoints'] > pointlimit:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints='$wWinner', points_added=0)
+							#self.current_rankings.append(new_ranking)
+						else:
+							new_ranking = dict(login=player['login'], nickname=player['name'], score=player['mappoints'], score_matchpoints=player['matchpoints'], points_added=0)
+							self.current_rankings.append(new_ranking)
```

Comment:
Same story here: can we move the 'mappoints' and 'map_points' saga into a variable?